### PR TITLE
feat: add ray orchestrator module

### PR DESCRIPTION
**Pull request by kukushking**
## Describe your changes
- added ray orchestrator module with a Step Function to orchestrate submission of Ray jobs via Kubernetes `BatchJob`
- added RBAC/identity mapping
- added test cases
- add sample `training-6B.py` script via Kubernetes `ConfigMap`
- update manifests
- update docs

## Checklist before requesting a review

- [x] I updated `CHANGELOG.MD` with a description of my changes
- [x] If the change was to a module, I ran the code validation script (`scripts/validate.sh`)
- [x] If the change was to a module, I have added thorough tests
- [x] If the change was to a module, I have added/updated the module's README.md
- [x] If a module was added, I added a reference to the module to the repository's README.md
- [x] I verified that my code deploys successfully using `seedfarmer apply`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 9fdf03bd41cbf306f30f3d3a9da21162d6b6d8a3
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=083780e3-f283-4580-90d9-0b7467832684%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 487f40c8c415fd839440afee5067db259c19291f
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=51fff48d-9a95-49c1-8eb1-38551e1ad28c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 3e3a37596c0dbec0be9f501c13b2f91c07cad59c
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=1a6e17c3-6842-4eeb-b8f8-3447dd23c611%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 9edd9b38f6efbb21207ebb0126b6a91fa580dd35
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=08dc1af1-2e40-4125-b66b-be2dbc58a8c8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: ca0804b0641b1b1153da205cba8c5b186c3f3417
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=cf98b87c-936b-4be2-9430-8e7296f5ca7b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 7573e8c01ad3e13099f68dcc1a8fb0b9f9860434
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=597b00fa-7c7c-4bed-ab13-f1372e90a7b7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/eks/ray-orchestrator/scripts/training-6B.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
+import os
```
Comment (kukushking) [inline modules/eks/ray-orchestrator/scripts/training-6B.py:1]: Considering to inject this via `dataFiles` altogether.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: dbff0cbb488617221b421ae1ae7f8f6801cb132e
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=32546fa4-37a8-4433-bb3f-2d1b8fe1a653%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/eks/ray-orchestrator/README.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
+# Ray Orchestrator
+
+## Description
+
+This module orcehstrates submission of a training job to the Ray Cluster using AWS Step Functions.
```
Comment (clokep) [inline modules/eks/ray-orchestrator/README.md:5]:
```suggestion
This module orchestrates submission of a training job to the Ray Cluster using AWS Step Functions.
```

File: modules/eks/ray-orchestrator/README.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+- `eks_cluster_admin_role_arn`- ARN of EKS admin role to authenticate kubectl
+- `eks_handler_role_arn`- ARN of EKS admin role to authenticate kubectl
+- `eks_cluster_name` - Name of the EKS cluster to deploy to
+- `eks_cluster_endpoint` - EKS cluster endpoint
+- `eks_oidc_arn` - ARN of EKS OIDC provider for IAM roles
+- `eks_cert_auth_data` - Auth certificate
+
+#### Optional
+
+- `step_function_timeout` - Step function timeout in minutes
```
Comment (clokep) [inline modules/eks/ray-orchestrator/README.md:23]: Does this have a default if undefined?

File: modules/eks/ray-orchestrator/modulestack.yaml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+              - "iam:UpdateRole"
+              - "iam:CreatePolicyVersion"
+              - "iam:DeletePolicyVersion"
+              - "iam:TagRole"
+              - "iam:UntagRole"
+              - "iam:PassRole"
+            Effect: Allow
+            Resource:
+              - !Sub "arn:aws:iam::${AWS::AccountId}:role/${ProjectName}-${DeploymentName}-${ModuleName}*"
+              - !Sub "arn:aws:iam::${AWS::AccountId}:role/service-role/${ProjectName}-${DeploymentName}-${ModuleName}*"
```
Comment (clokep) [inline modules/eks/ray-orchestrator/modulestack.yaml:50]: Not sure if it matters, but these should probably be partition agnostic?

File: modules/eks/ray-orchestrator/ray_orchestrator_stack.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        cluster = eks.Cluster.from_cluster_attributes(
+            self,
+            "EKSCluster",
+            cluster_name=eks_cluster_name,
+            open_id_connect_provider=provider,
+            kubectl_role_arn=eks_admin_role_arn,
+            kubectl_layer=KubectlV29Layer(self, "Kubectlv29Layer"),
+        )
+
+        with open(os.path.join(project_dir, "scripts/training-6B.py"), "r") as f:
```
Comment (clokep) [inline modules/eks/ray-orchestrator/ray_orchestrator_stack.py:69]: We had some other PR that forced things to lower-case, but I'm forgetting why. Will this be an issue here?

File: modules/eks/ray-orchestrator/README.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+- `data_bucket_name` - Name of the bucket to grant service account permissions to
+- `tags` - List of additional tags to apply to all resources
```
Comment (clokep) [inline modules/eks/ray-orchestrator/README.md:25]: I think tags is a map of string to string?

**Review by clokep — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 1445fab038db3136f2702262113ea1c6c01aa151
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=25bd21cc-2f1a-4c15-81ab-d22ab509dd33%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/eks/ray-orchestrator/README.md (hunk 3 above)
Comment (kukushking) [reply modules/eks/ray-orchestrator/README.md]: Thanks, updated.

File: modules/eks/ray-orchestrator/modulestack.yaml (hunk 4 above)
Comment (kukushking) [reply modules/eks/ray-orchestrator/modulestack.yaml]: It does - thanks. Updated.

File: modules/eks/ray-orchestrator/README.md (hunk 6 above)
Comment (kukushking) [reply modules/eks/ray-orchestrator/README.md]: Thanks, updated.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: bb44186d777bd6e912b14d3ca208a62489fb5e24
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=ef277d14-1f52-4487-935f-9720de60651d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by clokep — APPROVED** [review]

**Review by dgraeber — APPROVED** [review]

**Pull request merged**